app/service/rest_svc.py

```diff
@@ -438,6 +451,25 @@ async def _persist_source(self, access, source):
         await self._services.get('data_svc').load_source_file(file_path, allowed)
         return [s.display for s in await self._services.get('data_svc').locate('sources', dict(id=final['id']))]
 
+    async def _persist_objective(self, access, objective):
```

Comment:
there are few different distinct sections of this function that can be broken out to smaller functions